.github/workflows/build-deb-repo.yml

```diff
@@ -0,0 +1,149 @@
+        required: true
+        description: The Salt version to set prior to building packages.
+      nightly-build:
+        type: boolean
+        default: false
+      rc-build:
+        type: boolean
+        default: false
+      environment:
+        type: string
+        description: On which GitHub Environment Context To Run
+    secrets:
+      SECRETS_KEY:
+        required: true
+
+env:
+  COLUMNS: 160
+
+jobs:
+  build-repo:
+    name: DEB
+    environment: ${{ inputs.environment }}
+    runs-on:
+      - self-hosted
+      - linux
+      - repo-${{ inputs.environment }}
+    strategy:
+      fail-fast: false
+      max-parallel: 2
+      matrix:
+        include:
+          - distro: debian
+            version: "10"
+            arch: x86_64
+            key-id: 0E08A149DE57BFBE
+          - distro: debian
+            version: "10"
+            arch: aarch64
+            key-id: 0E08A149DE57BFBE
+          - distro: debian
+            version: "11"
+            arch: x86_64
+            key-id: 0E08A149DE57BFBE
+          - distro: debian
+            version: "11"
+            arch: aarch64
+            key-id: 0E08A149DE57BFBE
+          - distro: ubuntu
+            version: "18.04"
+            arch: x86_64
+            key-id: 0E08A149DE57BFBE
+          - distro: ubuntu
+            version: "20.04"
+            arch: x86_64
+            key-id: 0E08A149DE57BFBE
+          - distro: ubuntu
+            version: "20.04"
+            arch: aarch64
+            key-id: 0E08A149DE57BFBE
+          - distro: ubuntu
+            version: "22.04"
+            arch: x86_64
+            key-id: 0E08A149DE57BFBE
+          - distro: ubuntu
+            version: "22.04"
+            arch: aarch64
+            key-id: 0E08A149DE57BFBE
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Download System Dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y devscripts apt-utils
+
+      - name: Setup Python Tools Scripts
+        uses: ./.github/actions/setup-python-tools-scripts
+
+      - name: Download DEB Packages
+        uses: actions/download-artifact@v3
+        with:
+          name: salt-${{ inputs.salt-version }}-${{ matrix.arch }}-deb-pkgs
+          path: artifacts/pkgs/incoming
+
+      - name: Setup GnuPG
+        run: |
+          sudo install -d -m 0700 -o "$(id -u)" -g "$(id -g)" /run/gpg
+          GNUPGHOME="$(mktemp -d -p /run/gpg)"
+          echo "GNUPGHOME=${GNUPGHOME}" >> "$GITHUB_ENV"
+          cat <<EOF > "${GNUPGHOME}/gpg.conf"
+          batch
+          no-tty
+          pinentry-mode loopback
+          EOF
+
+
+      - name: Get Secrets
+        env:
+          SECRETS_KEY: ${{ secrets.SECRETS_KEY }}
+        run: |
+          SECRETS_KEY_FILE=$(mktemp /tmp/output.XXXXXXXXXX)
+          echo "$SECRETS_KEY" > "$SECRETS_KEY_FILE"
+          aws --region us-west-2 secretsmanager get-secret-value --secret-id /cmbu-saltstack/signing/repo-signing-keys \
```

Comment:
Question: These don't get logged anywhere when they return do they?

Reply:
No, they do not.
That workflow is also "protected" because in runs on a schedule, never pull requests, and, on top of that, it's tied to a GitHub Environment which has an allow list for which branches that environment is allowed to be used.

So, someone making changes on a pull requests to try and find out about the secrets will have a hard time doing so, and if they do succeed, it likely means we messed up a code review somewhere.

Reply:
If someone tries to use the environment, their job will fail unless its running on an allowed branch, which we are being quite strict.